pkg/trtl/honu.go

```diff
@@ -111,8 +112,74 @@ func (h *HonuService) Iter(ctx context.Context, in *pb.IterRequest) (out *pb.Ite
 	return nil, status.Error(codes.Unimplemented, "not implemented")
 }
 
-func (h *HonuService) Batch(stream pb.Trtl_BatchServer) (err error) {
-	return status.Error(codes.Unimplemented, "not implemented")
+// Batch is a client-side streaming request to issue multiple commands, usually Put and Delete.
+func (h *HonuService) Batch(stream pb.Trtl_BatchServer) error {
+	out := &pb.BatchReply{}
+	for {
+		// Read the next request from the stream.
+		in, err := stream.Recv()
+		if err == io.EOF {
+			return stream.SendAndClose(out)
+		}
+		if err != nil {
+			return status.Error(codes.Internal, err.Error())
+		}
+
+		out.Operations++
+		if in.Request == nil {
+			out.Failed++
+			out.Errors = append(out.Errors, &pb.BatchReply_Error{
+				Id:    in.Id,
+				Error: "missing request field",
+			})
+			continue
+		}
+
+		// Process the request.
+		switch in.Request.(type) {
+		case *pb.BatchRequest_Put:
+			var reply *pb.PutReply
+			if reply, err = h.Put(stream.Context(), in.Request.(*pb.BatchRequest_Put).Put); err != nil || !reply.Success {
```

Comment:
I think it's a bit simpler to replace `in.Request.(*pb.BatchRequest_Put).Put` with `in.GetPut()` - which should be available from the generated code. That method does essentially what you're doing (also with nil checks), but is a bit more readable. 

Also, I like that you're reusing the `h.Put` RPC call, but note that in the future we might have to break that call into two, an outer `h.Put` that handles thread safety and an inner `h.put` that can be called from either `Batch`, `Sync`, or `Put`.